utils/python/CIME/case.py

```diff
@@ -378,6 +378,27 @@ def _get_component_config_data(self):
             if result is not None:
                 del self.lookups[key]
 
+    def get_components(self):
+        # return dictionary of the form [component_class:component],
```

Comment:
These first two comments should be in docstring form:

    """
    doc
    """